test/BrickFarming-Unit.test.waffle.chai.ethers.js

```diff
@@ -0,0 +1,314 @@
+        });
+    });
+
+    describe("Do rewarding", function () {
+        it("deposit reward", async function () {
+            await farming.depositReward(gWei(10)); // 10 * 10 ^ 9
+            expect(await rewardToken.balanceOf(farming.address)).to.equal(
+                gWei(10)
+            );
+        });
+
+        it("check poolInfo", async function () {
+            const pool = await farming.poolInfo(0);
+
+            // AllocPoint: 100%
+            // Staked: 20 * 10 ^ 18
+            // DepositedReward: 10 * 10 ^ 9
+            // AccTokenPerShare: (10 * 10 ^ 9) * (10 ^ 12) / (20 * 10 ^ 18) = 500
+            expect(pool.accTokenPerShare).to.equal(500);
+        });
+
+        it("deposit reward", async function () {
+            await farming.depositReward(gWei(10)); // 10 * 10 ^ 9
+            expect(await rewardToken.balanceOf(farming.address)).to.equal(
+                gWei(20)
+            );
+        });
+
+        it("check poolInfo", async function () {
+            const pool = await farming.poolInfo(0);
+
+            expect(pool.accTokenPerShare).to.equal(1000);
+        });
+    });
+
+    describe("check claim", function () {
+        it("claim rewards for tester1", async () => {
+            await farming.connect(tester1).deposit(0, ZERO);
+        });
+
+        it("check poolInfo after tester1's claim", async function () {
+            const pool = await farming.poolInfo(0);
+
+            expect(pool.accTokenPerShare).to.equal(1000);
+            expect(pool.totalAmount).to.equal(ether(20));
+        });
+
+        it("check tester1 info", async function () {
+            const userInfo = await farming.userInfo(0, tester1.address);
+
+            expect(userInfo.amount).to.equal(ether(10));
+            expect(userInfo.rewardDebt).to.equal(gWei(10));
+            expect(userInfo.pendingRewards).to.equal(ZERO);
+
+            const userRewardTokenBalance = await rewardToken.balanceOf(
+                tester1.address
+            );
+
+            expect(userRewardTokenBalance).to.equal(gWei(10));
+        });
+
+        it("claim rewards for tester2", async () => {
+            await farming.connect(tester2).deposit(0, ZERO);
+        });
+
+        it("check poolInfo after tester2's claim", async function () {
+            const pool = await farming.poolInfo(0);
+
+            expect(pool.accTokenPerShare).to.equal(1000);
+            expect(pool.totalAmount).to.equal(ether(20));
+        });
+
+        it("check tester2 info", async function () {
+            const userInfo = await farming.userInfo(0, tester2.address);
+
+            expect(userInfo.amount).to.equal(ether(10));
+            expect(userInfo.rewardDebt).to.equal(gWei(10));
+            expect(userInfo.pendingRewards).to.equal(ZERO);
+
+            const userRewardTokenBalance = await rewardToken.balanceOf(
+                tester2.address
+            );
+
+            expect(userRewardTokenBalance).to.equal(gWei(10));
```

Comment:
@bejaxer you should also test that when the 3rd user deposits after we deposit reward token, the 3rd user's reward debt is his amount times the acc token per share so that he cannot claim *anything*